app/controllers/groups_controller.rb

```diff
@@ -199,4 +210,17 @@ def set_group
   def group_params
     params.require(:group).permit(:name)
   end
+
+  def expiration_date_param
+    expiration_date = params[:expiration_date]
+    return nil if expiration_date.nil? || expiration_date.empty?
+
+    Date.parse(expiration_date, '%Y-%m-%d')
+  end
+
+  def form_response(message)
```

Comment:
this method only use once and the name not showing intent, 
maybe we should just put the logic back into the main body of the caller

Reply:
method inlined